### PR TITLE
Upgrade to matrix-appservice-bridge 5.0.0

### PR DESCRIPTION
**Pull request by Half-Shot**
This includes a minimum Node version bump to Node.JS 16.0. This PR includes a handful of typing cleanups from where we were overusing the `!` operator, or ignoring obviously-wrong types.

### CODE REVIEW

File: src/AdminCommands.ts [hunk 1]
```diff
@@ -52,8 +52,12 @@ export class AdminCommands {
             this.yargs = this.yargs.command(
                 cmd.command,
                 cmd.description,
-                (yg: yargs.Argv) => void yg.options(cmd.options || {}),
-                cmd.handler.bind(cmd)
+                cmd.options ?? {},
+                // Our intergration with yargs is quite broken and messy, so
+                // to avoid further pain this just assunes that the handler/yargs
```
Comment (AndrewFerr) [inline src/AdminCommands.ts:57]: Nitpick, "assunes" -> "assumes"

File: tests/integration/HttpTests.ts [hunk 2]
```diff
@@ -29,7 +29,7 @@ describe("HttpTests", () => {
 
     it("will respond 200 to a health check", async () => {
         await harness.main.run(57000);
-        const res = await axios.get("http://localhost:57000/health");
+        const res = await axios.get("http://127.0.0.1:57000/health");
```
Comment (AndrewFerr) [inline tests/integration/HttpTests.ts:32]: Out of curiosity, what broke without this?

File: package.json [hunk 3]
```diff
@@ -3,15 +3,15 @@
   "version": "1.11.0",
   "description": "A Matrix <--> Slack bridge",
   "engines": {
-    "node": ">=12 16<="
+    "node": ">=16 18<="
```
Comment (AndrewFerr) [inline package.json:6]: `>=16 <=18`

**Review by AndrewFerr — COMMENTED** [review]
I just skimmed through the typing cleanups, but they look fine overall.  Lint warnings are also down a bit (-13), which is nice.

**Review by tadzik — CHANGES_REQUESTED** [review]
Needs fixes to the Dockerfile, as it currently fails to start due to:

```
Error: Cannot find module '@matrix-org/matrix-sdk-crypto-nodejs-linux-x64-gnu'
```

Dropping `--ignore-scripts` fixes it

**Review by tadzik — APPROVED** [review]

**Pull request merged**